util/Migrator/DbScripts/2023-07-03_00_SecretsManagerBetaColumn.sql

```diff
@@ -0,0 +1,382 @@
+        [UseSecretsManager] = @UseSecretsManager,
+        [Status] = @Status,
+        [UsePasswordManager] = @UsePasswordManager,
+        [SmSeats] = @SmSeats,
+        [SmServiceAccounts] = @SmServiceAccounts,
+        [MaxAutoscaleSmSeats] = @MaxAutoscaleSmSeats,
+        [MaxAutoscaleSmServiceAccounts] = @MaxAutoscaleSmServiceAccounts,
+        [SecretsManagerBeta] = @SecretsManagerBeta
+    WHERE
+        [Id] = @Id
+END
+GO
+
+-- Add column `SecretsManagerBeta` in OrganizationUserOrganizationDetailsView
+CREATE OR ALTER VIEW [dbo].[OrganizationUserOrganizationDetailsView]
+AS
+SELECT
+    OU.[UserId],
+    OU.[OrganizationId],
+    O.[Name],
+    O.[Enabled],
+    O.[PlanType],
+    O.[UsePolicies],
+    O.[UseSso],
+    O.[UseKeyConnector],
+    O.[UseScim],
+    O.[UseGroups],
+    O.[UseDirectory],
+    O.[UseEvents],
+    O.[UseTotp],
+    O.[Use2fa],
+    O.[UseApi],
+    O.[UseResetPassword],
+    O.[SelfHost],
+    O.[UsersGetPremium],
+    O.[UseCustomPermissions],
+    O.[UseSecretsManager],
+    O.[Seats],
+    O.[MaxCollections],
+    O.[MaxStorageGb],
+    O.[Identifier],
+    OU.[Key],
+    OU.[ResetPasswordKey],
+    O.[PublicKey],
+    O.[PrivateKey],
+    OU.[Status],
+    OU.[Type],
+    SU.[ExternalId] SsoExternalId,
+    OU.[Permissions],
+    PO.[ProviderId],
+    P.[Name] ProviderName,
+    P.[Type] ProviderType,
+    SS.[Data] SsoConfig,
+    OS.[FriendlyName] FamilySponsorshipFriendlyName,
+    OS.[LastSyncDate] FamilySponsorshipLastSyncDate,
+    OS.[ToDelete] FamilySponsorshipToDelete,
+    OS.[ValidUntil] FamilySponsorshipValidUntil,
+    OU.[AccessSecretsManager],
+    O.[UsePasswordManager],
+    O.[SmSeats],
+    O.[SmServiceAccounts],
+    O.[SecretsManagerBeta]
+FROM
+    [dbo].[OrganizationUser] OU
+LEFT JOIN
+    [dbo].[Organization] O ON O.[Id] = OU.[OrganizationId]
+LEFT JOIN
+    [dbo].[SsoUser] SU ON SU.[UserId] = OU.[UserId] AND SU.[OrganizationId] = OU.[OrganizationId]
+LEFT JOIN
+    [dbo].[ProviderOrganization] PO ON PO.[OrganizationId] = O.[Id]
+LEFT JOIN
+    [dbo].[Provider] P ON P.[Id] = PO.[ProviderId]
+LEFT JOIN
+    [dbo].[SsoConfig] SS ON SS.[OrganizationId] = OU.[OrganizationId]
+LEFT JOIN
+    [dbo].[OrganizationSponsorship] OS ON OS.[SponsoringOrganizationUserID] = OU.[Id]
+GO
+
+-- Update the metadata for the following stored procedures and view so that the SecretsManagerBeta column is included
+IF OBJECT_ID('[dbo].[OrganizationUserOrganizationDetails_ReadByUserIdStatus]') IS NOT NULL
+BEGIN
+    EXECUTE sp_refreshsqlmodule N'[dbo].[OrganizationUserOrganizationDetails_ReadByUserIdStatus]';
+END
+GO
+
+IF OBJECT_ID('[dbo].[OrganizationUserOrganizationDetails_ReadByUserIdStatusOrganizationId]') IS NOT NULL
+BEGIN
+    EXECUTE sp_refreshsqlmodule N'[dbo].[OrganizationUserOrganizationDetails_ReadByUserIdStatusOrganizationId]';
+END
+GO
+
+IF OBJECT_ID('[dbo].[OrganizationView]') IS NOT NULL
+BEGIN
+    EXECUTE sp_refreshsqlmodule N'[dbo].[OrganizationView]';
```

Comment:
For views we should use `sp_refreshview` ([documentation](https://contributing.bitwarden.com/contributing/code-style/sql#adjusting-metadata))
```suggestion
    EXECUTE sp_refreshview N'[dbo].[OrganizationView]';
```

Reply:
Also, because `[dbo].[OrganizationView]` will have a new column, shouldn't we execute `sp_refreshsqlmodule` for all  stored procedures that use it? Or does `sp_refreshview` already take care of that?

These are some stored procedures that select all columns from `[dbo].[OrganizationView]`:
- `Organization_ReadByUserId`
- `Organization_ReadByEnabled`
- `Organization_ReadById`
- `Organization_ReadByIdentifier`
- `Organization_ReadByLicenseKey`
- `Organization_ReadByProviderId`
- `Organization_Search`
- `Organization_UnassignedToProviderSearch`

Reply:
I'm almost certain that by running `EXECUTE sp_refreshview N'[dbo].[OrganizationView]';` all those dependant stored procedures will be updated but I just want to be sure.

Reply:
I switched to using `sp_refreshview`, thanks for pointing that out!

Sadly the MS docs for [sp_refreshview](https://learn.microsoft.com/en-us/sql/relational-databases/system-stored-procedures/sp-refreshview-transact-sql) don't seem to answer the question you asked. It's a great question and I'm curious myself. The view's metadata is updated, but what about objects that depend on that view? By the same logic, you would think objects that depend on that view need to be updated (just as the view needs an update because the table it depends on was changed).

However, through executing some of the above stored procedures manually, the SecretsManagerBeta column _is_ present.